internal/http/services/owncloud/ocdav/put.go

```diff
@@ -249,8 +281,21 @@ func (s *svc) handlePutHelper(w http.ResponseWriter, r *http.Request, content io
 				w.WriteHeader(http.StatusPartialContent)
 				return
 			}
+			if httpRes.StatusCode == errtypes.StatusChecksumMismatch {
+				w.WriteHeader(http.StatusBadRequest)
+				xml := `<?xml version="1.0" encoding="utf-8"?>
+<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
+  <s:exception>Sabre\DAV\Exception\BadRequest</s:exception>
+  <s:message>The computed checksum does not match the one received from the client.</s:message>
+</d:error>`
```

Comment:
I believe we have a package for this kind of errors: https://github.com/cs3org/reva/blob/bd7234e5b29361f8a0047f620c72671aeca0ded7/internal/http/services/owncloud/ocdav/error.go, in particular: https://github.com/cs3org/reva/blob/bd7234e5b29361f8a0047f620c72671aeca0ded7/internal/http/services/owncloud/ocdav/error.go#L48-L55